src/preferences/dialog/dlgprefcolors.cpp

```diff
@@ -227,18 +217,62 @@ void DlgPrefColors::slotHotcuePaletteChanged(const QString& paletteName) {
     comboBoxHotcueDefaultColor->setCurrentIndex(defaultColor);
 }
 
-void DlgPrefColors::slotEditClicked() {
-    pushButtonEdit->hide();
-    labelCustomPalette->hide();
-    widgetSpacer->hide();
-    groupBoxPaletteEditor->show();
+void DlgPrefColors::slotEditTrackPaletteClicked() {
+    QString trackColorPaletteName = comboBoxTrackColors->currentText();
+    openColorPaletteEditor(trackColorPaletteName, false);
 }
 
-void DlgPrefColors::slotCloseClicked() {
-    groupBoxPaletteEditor->hide();
-    widgetSpacer->show();
-    pushButtonEdit->show();
-    labelCustomPalette->show();
+void DlgPrefColors::slotEditHotcuePaletteClicked() {
+    QString hotcueColorPaletteName = comboBoxHotcueColors->currentText();
+    openColorPaletteEditor(hotcueColorPaletteName, true);
+}
+
+void DlgPrefColors::openColorPaletteEditor(
+        const QString& paletteName,
+        bool editHotcuePalette) {
```

Comment:
I think the "Assign to hotcue number" column should be hidden in the dialog if this bool is false. It has no significance for the track colors.

Reply:
This is an issue of the shared palettes model. A palette has always the column and it should therefore always visible and accessible, regardless from which combobox the editor was started.
This flag is only used to automatically select the newly edited palette. That's all.

Reply:
The shared model is no problem. We can just hide the column from the QTableView. I implemented it here: https://github.com/daschuer/mixxx/pull/52

Reply:
I don't think we as should do this.

Reply:
What do others think? @Holzhaus, @ronso0?

Reply:
I don't have a strong opinion about this either way, but I think it would be a small improvement.

Reply:
Yeah let's hide it when a the editor is opened for track colors, even if it shows the same palettes as for hotcues.

Reply:
Yes, I think we can hide the column. If a palette is used for track colors, the "assign to hotcue number" is of no concern. The only case where this could be confusing is when the same palette is used for both comboboxes (track color and hotcues). We have three options:

1. Always show the column (both for track colors and hotcue colors)
2. Hide the column for track color palettes
3. Hide the column for track color palettes unless the track color palette is also used for hotcues

I'm okay with any of these options.

Reply:
I have just merged daschuer#52